tensorflow/stream_executor/cuda/cuda_gpu_executor.cc

```diff
@@ -1129,6 +1129,37 @@ DeviceDescription *CUDAExecutor::PopulateDeviceDescription() const {
           CU_DEVICE_ATTRIBUTE_MAX_REGISTERS_PER_MULTIPROCESSOR, device_)
           .ValueOrDie());
 
+  // We are loading a dummy ptx kernel to set the device description's
+  // blocks_per_core_limit by calling the CUDA occupancy calculator.  This
+  // value is currently required XLA GPU's CalculateLaunchDimensions()
+  const char* blank_ptx = R"(
+.version 6.0
+.target sm_30
+.address_size 64
+
+        // .globl       testkernel
+.visible .entry testkernel()
+{
+        ret;
+})";
+  const char* kernel_name = "testkernel";
+
+  CUmodule blank_module;
+  CUfunction blank_function;
+  CUDADriver::LoadPtx(context_, blank_ptx, &blank_module);
+  CUDADriver::GetModuleFunction(context_, blank_module, kernel_name,
+                                &blank_function);
+
+  int bpc;
+  CUresult result = cuOccupancyMaxActiveBlocksPerMultiprocessor(
+      &bpc, blank_function, 1, 1);
+  if (result != CUDA_SUCCESS) {
+    VLOG(1) << "Failed to calculate max blocks per SM using dummy kernel.";
```

Comment:
I think you can LOG(ERROR) this one.  VLOG will only be visible if the user explicitly asks for it.  Assuming this function only runs once (or once per GPU) a LOG(ERROR) is fine.  (I'd rather mildly annoy users and have them ask me why it's failing than have no idea that there's a bug here.)

Reply:
Sounds good
```suggestion
    LOG(ERROR) << "Failed to calculate max blocks per SM using dummy kernel.";
```